tests/update_workflow.go

```diff
@@ -3506,26 +3364,20 @@ func (s *FunctionalSuite) TestUpdateWorkflow_SpeculativeWorkflowTask_Heartbeat()
 	}()
 
 	// Heartbeat from workflow.
-	res, err := poller.PollAndProcessWorkflowTask(WithRetries(1), WithForceNewWorkflowTask)
+	res, err := poller.PollAndProcessWorkflowTask(WithoutRetries, WithForceNewWorkflowTask)
 	s.NoError(err)
 	heartbeatResp := res.NewTask
 
 	// Reject update from workflow.
-	updateResp, err := poller.HandlePartialWorkflowTask(heartbeatResp.GetWorkflowTask(), true)
+	updateResp, err := poller.HandlePartialWorkflowTask(heartbeatResp.GetWorkflowTask(), false)
 	s.NoError(err)
+	s.NotNil(updateResp)
 	updateResult := <-updateResultCh
 	s.Equal("rejection-of-"+tv.UpdateID("1"), updateResult.GetOutcome().GetFailure().GetMessage())
 	s.EqualValues(0, updateResp.ResetHistoryEventId, "no reset of event ID should happened after update rejection because of heartbeat")
 
-	// Complete workflow.
-	completeWorkflowResp, err := poller.HandlePartialWorkflowTask(updateResp.GetWorkflowTask(), false)
-	s.NoError(err)
-	s.NotNil(completeWorkflowResp)
-	s.Nil(completeWorkflowResp.GetWorkflowTask())
-	s.EqualValues(0, completeWorkflowResp.ResetHistoryEventId)
```

Comment:
I noticed a couple of these were removed. Is that because they didn't provide any value?

Reply:
Yes, no reason to actually complete WF and make unrelated asserts. I wanted to remove all of them, but in other places, that last WFT actually does some update related stuff, so I left it there. In some other places there are test for WF completion.